ui/app/components/auth/form/base.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+
+import type AuthService from 'vault/vault/services/auth';
+import type ClusterModel from 'vault/models/cluster';
+import type { HTMLElementEvent } from 'vault/forms';
+
+/**
+ * @module Auth::Base
+ *
+ * @param {string} authType - chosen login method type
+ * @param {object} cluster - The cluster model which contains information such as cluster id, name and boolean for if the cluster is in standby
+ * @param {function} onError - callback if there is a login error
+ * @param {function} onSuccess - calls onAuthResponse in auth/page redirects if successful
+ */
+
+interface Args {
+  authType: string;
+  cluster: ClusterModel;
+  onError: CallableFunction;
+  onSuccess: CallableFunction;
+}
+
+export default class AuthBase extends Component<Args> {
```

Comment:
For reference, much of the logic in this component is pulled from https://github.com/hashicorp/vault/blob/741721ad0e2f1888114941a3009133cccdc4b7b2/ui/app/components/auth/login-form.js